tensorflow/core/kernels/reduction_ops_gpu_kernels.h

```diff
@@ -238,7 +238,10 @@ __global__ void ColumnReduceMax16ColumnsKernel(
   if (row * num_cols + col < num_rows * num_cols)
     sum = in[row * num_cols + col];
 
-  __shared__ value_type partial_sums[32][33];
+  // 1D array necessary due to bug in CUDA 9 compiler.
+  //__shared__ value_type partial_sums[32][33];
```

Comment:
Can we change this to a TODO when the bug is fixed?  (Here and similar places)

How will we know when the bug is fixed?  Not until CUDA 10?  Does this bug break all 2D shared arrays?

Also, I think you need need a space after the // or the linter will complain.

Reply:
Not all 2D shared arrays are affected. As far as I can tell, only those whose types rely on constexpr default constructors enabled by NVCC's -expt-relaxed-constexpr option.